Scenes/Player/player.gd

```diff
@@ -37,44 +39,46 @@ func set_current_health(_current_health):
 	current_health = _current_health
 	# TODO: Check if player died. Perform on death actions
 
+
 func register(world):
-	world.add_child(hitbox)
+	world.add_child(world_player)
 	server = world.get_node("/root/Server")
 
 func update(new_state):
-  hitbox.position = new_state[si.PLAYER_POSITION]
+  world_player.position = new_state[si.PLAYER_POSITION]
 
 func remove():
-	if hitbox == null:
+	if world_player == null:
 		Logger.warn("Player disconnected before registering")
 	else:
-		var player_id = hitbox.id
+		var player_id = world_player.id
 		server.broadcast_packet(Players.get_players([player_id]),
 			si.create_player_despawn_packet(player_id))
-		hitbox.queue_free()
-		hitbox = null
+		world_player.queue_free()
+		world_player = null
 
 func get_position():
-	return hitbox.position
+	return world_player.position
 
 func mock_stats():
 	stats["current_health"] = 100
 	stats["max_health"] = 100
 	stats["attack"] = 150
 
-func take_damage(value, attacker):
-	var current_health =  stats["current_health"]
-	current_health = max(0, current_health - value)
-	stats["current_health"] = current_health
-	hitbox.display("Current health: " + str(current_health))
+func take_damage(damage_value, attacker):
+	stats["current_health"] -= damage_value
+	world_player.display("Current health: %d" % [stats["current_health"]])
 	server.broadcast_packet(Players.get_players(),
-		si.create_player_take_damage_packet(attacker, hitbox.id, value))
-
+		si.create_player_take_damage_packet(attacker, world_player.id, damage_value))
+	if stats["current_health"] <= 0:
+		world_player.position = Vector2(250, 250)
+		si.create_player_died_packet(world_player.id, world_player.position)
```

Comment:
This has no effect, it needs to be sent via `server.send_packet` or `server.broadcast_packet`